### PR TITLE
refactor: unify interleaved MRoPE rotary embedding

### PR DESCRIPTION
**Pull request by CUHKSZzxy**
## Summary

- unify MRoPE rotary table generation so Qwen3-VL interleaved MRoPE and Qwen2-style chunked MRoPE share the backend rotary embedding path
- keep text-only position ids on the regular RoPE path
- add focused coverage for chunked MRoPE, interleaved MRoPE, YaRN MRoPE overrides, and text-position fallback
- remove the MRoPE position-id padding helper and its dedicated test case

## Validation

- focused rotary embedding unit tests passed

## Assistance

Assisted with Codex + GPT-5.5 xHigh Fast

### CODE REVIEW

File: lmdeploy/pytorch/nn/rotary_embedding.py [hunk 1]
```diff
@@ -252,6 +273,107 @@ def forward(self, query: Tensor, key: Tensor, cos: Tensor, sin: Tensor, inplace:
...
+            axis = index % 3
+            selected_chunks.append(chunk[axis])
+        return torch.cat(selected_chunks, dim=-1)
+
+    def apply_interleaved_mrope(self, freqs: torch.Tensor):
+        """Apply Qwen3-VL style interleaved MRoPE."""
+        # Layout is lane-interleaved: T, H, W, T, H, W...; start from T and
+        # overwrite the H/W lanes from their corresponding axes.
+        half_dim = sum(self.mrope_section)
+        has_duplicated_half = freqs.size(-1) == half_dim * 2
```
[the remark this answers is not in the document]
Comment (CUHKSZzxy) [reply lmdeploy/pytorch/nn/rotary_embedding.py:325]: Thanks. For supported interleaved MRoPE configs, `sum(mrope_section)` is the first-half rotary dim and backend tables are either first-half freqs or duplicated cos/sin width. This helper intentionally stays lean; invalid `mrope_section` values are outside the supported model-config contract and should be handled at model/config support time rather than in the hot selection path.

File: lmdeploy/pytorch/nn/rotary_embedding.py [hunk 2]
```diff
@@ -252,6 +273,107 @@ def forward(self, query: Tensor, key: Tensor, cos: Tensor, sin: Tensor, inplace:
...
+
+    def apply_chunked_mrope(self, freqs: torch.Tensor):
+        """Apply Qwen2-VL style chunked MRoPE."""
+        # Layout is contiguous bands: T..., H..., W..., then repeated for the
+        # duplicated RoPE half if freqs already contains cos/sin table width.
+        mrope_section = self.mrope_section
+        if freqs.size(-1) == sum(self.mrope_section) * 2:
+            mrope_section = mrope_section * 2
+        selected_chunks = []
+        for index, chunk in enumerate(freqs.split(mrope_section, dim=-1)):
```
[the remark this answers is not in the document]
Comment (CUHKSZzxy) [reply lmdeploy/pytorch/nn/rotary_embedding.py:315]: Thanks. We keep this as a model config contract: `mrope_section` comes from the model RoPE parameters and must match the effective rotary dim after `partial_rotary_factor`. The focused tests cover that partial-rotary case. Adding runtime validation here would put defensive config checking on the rotary path without changing supported-model behavior.

**Review by grimoire — APPROVED** [review]
LGTM
